### PR TITLE
[core] Enable incremental vacuum for Offline DB

### PR DESCRIPTION
**Pull request by pozdnyakov**
Thus we avoid re-creating the whole database and keeping the backup file as it happens on calling VACUUM.

Fixes: https://github.com/mapbox/mapbox-gl-native-team/issues/78
Fixes: https://github.com/mapbox/mapbox-gl-native/issues/15728

### CODE REVIEW

File: platform/default/src/mbgl/storage/offline_database.cpp [hunk 1]
```diff
@@ -181,6 +187,16 @@ void OfflineDatabase::migrateToVersion6() {
     transaction.commit();
 }
 
+void OfflineDatabase::migrateToVersion7() {
+    assert(db);
+    if (getPragma<int64_t>("PRAGMA auto_vacuum") != 2) {
+        // auto_vacuum might have been already set to INCREMENTAL at migrateToVersion3()
+        db->exec("PRAGMA auto_vacuum = INCREMENTAL");
+        db->exec("VACUUM");
```
Comment (halset) [inline platform/default/src/mbgl/storage/offline_database.cpp]: If the user has a 1GB database on verion 6, will this `VACUUM` result in a copy of 1GB data? Could this lead to upgrade taking a long time?

Comment (pozdnyakov) [reply platform/default/src/mbgl/storage/offline_database.cpp]: This is a fair point, thanks! On the other hand assigning the `auto_vacuum` pragma to a new value is not a breaking change. So, after brainstorming with @tmpsantos and @alexshalamov, we think that we can just lazily reassign the `auto_vacuum` pragma for the existing databases even without version bumping (newly created databases will have `auto_vacuum = INCREMENTAL` from the start).

Comment (halset) [reply platform/default/src/mbgl/storage/offline_database.cpp]: @pozdnyakov Thanks! Could we skip `VACUUM` here or is it needed?

Comment (pozdnyakov) [reply platform/default/src/mbgl/storage/offline_database.cpp]: it's required, so that `auto_vacuum = INCREMENTAL` gets to an action.

**Pull request merged**